src/main/java/frc/team2412/robot/subsystems/LauncherSubsystem.java

```diff
@@ -0,0 +1,115 @@
+	private final SparkAbsoluteEncoder launcherAngleEncoder;
+	private final SparkPIDController launcherAnglePIDController;
+
+	private final GenericEntry launcherSpeedEntry =
+			Shuffleboard.getTab("Launcher")
+					.addPersistent("Launcher Speed", SPEAKER_SHOOT_SPEED)
+					.withSize(1, 1)
+					.withWidget(BuiltInWidgets.kNumberSlider)
+					.withProperties(Map.of("Min", -1, "Max", 1))
+					.getEntry();
+
+	private final GenericEntry launcherAngleEntry =
+			Shuffleboard.getTab("Launcher")
+					.addPersistent("Launcher angle", getAngle())
+					.withSize(1, 1)
+					.withWidget(BuiltInWidgets.kTextView)
+					.getEntry();
+	// Constructor
+	public LauncherSubsystem() {
+
+		// MOTOR INSTANCE VARIBLES
+		// motors
+		launcherTopMotor = new CANSparkFlex(Hardware.LAUNCHER_TOP_MOTOR_ID, MotorType.kBrushless);
+		launcherBottomMotor = new CANSparkFlex(Hardware.LAUNCHER_BOTTOM_MOTOR_ID, MotorType.kBrushless);
+		launcherAngleMotor = new CANSparkFlex(Hardware.LAUNCHER_ANGLE_MOTOR_ID, MotorType.kBrushless);
+		// encoders
+		launcherTopEncoder = launcherTopMotor.getEncoder();
+		launcherBottomEncoder = launcherBottomMotor.getEncoder();
+		launcherAngleEncoder = launcherAngleMotor.getAbsoluteEncoder(Type.kDutyCycle);
+
+		// PID controllers
+		launcherAnglePIDController = launcherAngleMotor.getPIDController();
+		launcherAnglePIDController.setFeedbackDevice(launcherAngleEncoder);
+	}
+
+	public void configMotors() {
+		launcherTopMotor.restoreFactoryDefaults();
+		launcherBottomMotor.restoreFactoryDefaults();
+		launcherAngleMotor.restoreFactoryDefaults();
+		// idle mode (wow)
+		launcherTopMotor.setIdleMode(IdleMode.kCoast);
+		launcherBottomMotor.setIdleMode(IdleMode.kCoast);
+		launcherAngleMotor.setIdleMode(IdleMode.kBrake);
+		// inveritng the bottom motor lmao
+		launcherBottomMotor.setInverted(true);
+
+		// current limit
+		launcherTopMotor.setSmartCurrentLimit(20);
+		launcherBottomMotor.setSmartCurrentLimit(20);
+		launcherAngleMotor.setSmartCurrentLimit(20);
+
+		launcherTopMotor.burnFlash();
+		launcherBottomMotor.burnFlash();
+		launcherAngleMotor.burnFlash();
+	}
+
+	// stop specific motor method
+	public void stopLauncher() {
+		launcherTopMotor.set(0);
+		launcherBottomMotor.set(0);
+	}
+
+	public void shoot() {
+		launcherTopMotor.set(launcherSpeedEntry.getDouble(SPEAKER_SHOOT_SPEED));
+		launcherBottomMotor.set(launcherSpeedEntry.getDouble(SPEAKER_SHOOT_SPEED));
+	}
+	// returns the degrees of the angle of the launcher
+	public double getAngle() {
+		// get position returns a double in the form of rotations
+		return Units.rotationsToDegrees(launcherAngleEncoder.getPosition());
+	}
+
+	public void setAngle(double angle) {
+		launcherAnglePIDController.setReference(Units.degreesToRotations(angle), ControlType.kPosition);
+	}
+
+	@Override
+	public void periodic() {
+		// .get will be replaced with .getVelocity once PID is established for flywheels :C
+		launcherSpeedEntry.setDouble(launcherTopMotor.get());
```

Comment:
When the launcher stops, I think this will set the launcher setpoint used by `shoot()` to zero, which is unexpected. It'd be better to have separate dashboard entries for the current speed/power, and the desired speed/power used when `shoot()` is called.

Reply:
How would i get the current speed then?

Reply:
i think i figured this out, ty!